src/ProjectOrigin.WalletSystem.IntegrationTests/REST/WalletControllerTests.cs

```diff
@@ -0,0 +1,158 @@
+using System;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using AutoFixture;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using ProjectOrigin.WalletSystem.IntegrationTests.TestClassFixtures;
+using ProjectOrigin.WalletSystem.IntegrationTests.TestExtensions;
+using ProjectOrigin.WalletSystem.Server.Database;
+using ProjectOrigin.WalletSystem.Server.Services.REST.v1;
+using Xunit;
+
+namespace ProjectOrigin.WalletSystem.IntegrationTests;
+
+public class WalletControllerTests : IClassFixture<PostgresDatabaseFixture>
+{
+    private readonly Fixture _fixture;
+    private readonly PostgresDatabaseFixture _dbFixture;
+    private readonly IUnitOfWork _unitOfWork;
+
+    public WalletControllerTests(PostgresDatabaseFixture postgresDatabaseFixture)
+    {
+        _fixture = new Fixture();
+        _dbFixture = postgresDatabaseFixture;
+        _unitOfWork = _dbFixture.CreateUnitOfWork();
+    }
+
+    [Theory]
+    [InlineData("Europe/Copenhagen", new long[] { 1000, 2400, 1400 }, CertificateType.Production)]
+    [InlineData("Europe/London", new long[] { 110, 240, 130 }, CertificateType.Consumption)]
+    [InlineData("America/Toronto", new long[] { 1600, 2400, 800 }, CertificateType.Production)]
+    public async Task Test_AggregateCertificates(string timezone, long[] values, CertificateType type)
+    {
+        // Arrange
+        var issuestartDate = new DateTimeOffset(2020, 6, 1, 12, 0, 0, TimeSpan.Zero);
+        var issueEndDate = new DateTimeOffset(2020, 6, 30, 12, 0, 0, TimeSpan.Zero);
+        var queryStartDate = new DateTimeOffset(2020, 6, 8, 12, 0, 0, TimeSpan.Zero);
+        var queryEndDate = new DateTimeOffset(2020, 6, 10, 12, 0, 0, TimeSpan.Zero);
+
+        var subject = _fixture.Create<string>();
+        var controller = new WalletController
```

Comment:
Why are you testing test the controller directly and not with a integration/api/flow test?

Reply:
Faster executing tests, and less pieces that can go wrong.